email/1.3/modules/ROOT/edits/email-trigger.adoc

```diff
@@ -0,0 +1,219 @@
+
+The following example shows how to configure these parameters in Anypoint Studio by dispatching all messages that contain the word `IMPORTANT`:
+
+. Select the *On New Email - IMAP* or *On New Email - POP3* source in the Studio canvas.
+. On the source configuration screen, for the *Imap matcher* or *Pop3 matcher* parameters, select either:
+
+* *Edit inline* +
+Set all the inline fields *Seen*, *Answered*, *Deleted*, *Recent*, and so on.
+* *Global reference* +
+Create a global reference configuration to set all the inline fields.
+
+[start=3]
+. In the *Subject regex* field, set the value as `IMPORTANT`.
+
+The following screenshot shows the *Imap matcher*  edit inline configuration for the *On New Email - IMAP* source:
+
+.Imap matcher with On New Email - IMAP source
+image::email-imap-matcher.png[Imap matcher with On New Email - IMAP]
+
+In the XML editor, the configuration `imap-matcher subjectRegex="IMPORTANT"` is under the `listener-imap` tag:
+
+[source, xml, linenums]
+----
+<flow name="OnNewEmail-IMAP">
+    <email:listener-imap config-ref="Email_IMAP">
+        <scheduling-strategy>
+            <fixed-frequency/>
+        </scheduling-strategy>
+        <email:imap-matcher subjectRegex="IMPORTANT"/>
+    </email:listener-imap>
+</flow>
+----
+
+== Configure Attachment Naming Strategy Parameter
+
+Email Connector *On New Email - IMAP*, *On New Email - POP3*, *List - IMAP*, and *List - POP3* operations support the *Attachment naming strategy* parameter, which enables you to specify a strategy for naming attachments. You can set the parameter at the global element or override it at the operation level. The parameter has the following options to choose from:
+
+* *NAME (Default)* +
+Email Connector names the attachment after the value of the `filename` field in the `Content-Disposition` line of the attachment (this is the standard and recommended option to set the name for an attachment). If the `filename` field is not present or is empty, then the attachment is named `Unnamed`.
+If multiple attachments have no name, they are called `Unnamed`, `Unnamed_1`, `Unnamed_2`, and so on, so that they are addressed individually.
+* *NAME_HEADERS* +
+Email Connector first tries to obtain the name from the `filename` field in the `Content-Disposition` line (the same as when using `NAME`). If that fails, the connector looks for the `name` header in the `Content-Type` line of the attachment. If the `name` header is not present or is empty, then the attachment is named `Unnamed`.
+* *NAME_HEADERS_SUBJECT* +
+Email Connector tries the two previously described options. If both fail, then the connector checks if the attachment itself is an email (a nested message) and looks for the subject of the email. If the subject is not empty, then the attachment is named after the subject. Otherwise, the attachment is named `Unnamed`.
```

Comment:
```suggestion
If both *NAME_HEADERS* and *NAME_HEADERS_SUBJECT* fail, then the connector checks if the attachment itself is an email (a nested message) and looks for the subject of the email. If the subject is not empty, then the attachment is named after the subject. Otherwise, the attachment is named `Unnamed`.
```

Reply:
BUT how do the two previous fail, if they automatically end up with 'Unnamed"?